app/models.py

```diff
@@ -218,6 +218,45 @@ def set_fully_registered(self):
             return
         db.session.add(UserJoinedEvent.from_user(self))
 
+    def match(self, level, limit=10):
+        '''
+        Return a list of users that can answer questions this user wants to
+        learn at the specified level.
```

Comment:
Alternatively, maybe the function could be refactored into smaller functions that are composed with one another? Might make it a bit easier to understand... or maybe I'm just dense, hehe.